### PR TITLE
SSL support for Mattermost in Botkube

### LINKED ISSUE #81: [BUG] mattermost self-signed certificate
**Describe the bug**
I have private mattermost server with self-signed certificate. I'm tried to add custom CA to docker container.

```
INFO[2019-04-16T06:44:07Z] Allowed Events - map[{Resource:clusterrole Namespace:all}:true {Resource:configmap Namespace:all}:true {Resource:daemonset Namespace:all}:true {Resource:deployment Namespace:all}:true {Resource:ingress Namespace:all}:true {Resource:job Namespace:all}:true {Resource:namespace Namespace:all}:true {Resource:node Namespace:all}:true {Resource:persistentvolume Namespace:all}:true {Resource:persistentvolumeclaim Namespace:all}:true {Resource:pod Namespace:all}:true {Resource:role Namespace:all}:true {Resource:rolebinding Namespace:all}:true {Resource:secret Namespace:all}:true {Resource:service Namespace:all}:true]
DEBU[2019-04-16T06:44:07Z] Registering the filter&{}
DEBU[2019-04-16T06:44:07Z] Registering the filter&{}
DEBU[2019-04-16T06:44:07Z] Registering the filter&{}
INFO[2019-04-16T06:44:07Z] Starting controller
INFO[2019-04-16T06:44:07Z] Starting mattermost bot
panic: runtime error: invalid memory address or nil pointer dereference
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x115de22]

goroutine 53 [running]:
github.com/infracloudio/botkube/vendor/github.com/mattermost/mattermost-server/model.(*WebSocketClient).Listen.func1.1(0x0)
        /go/src/github.com/infracloudio/botkube/vendor/github.com/mattermost/mattermost-server/model/websocket_client.go:101 +0x22
panic(0x1314c20, 0x2322130)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/infracloudio/botkube/vendor/github.com/mattermost/mattermost-server/model.(*WebSocketClient).Listen.func1(0x0)
        /go/src/github.com/infracloudio/botkube/vendor/github.com/mattermost/mattermost-server/model/websocket_client.go:109 +0x64
created by github.com/infracloudio/botkube/vendor/github.com/mattermost/mattermost-server/model.(*WebSocketClient).Listen
        /go/src/github.com/infracloudio/botkube/vendor/github.com/mattermost/mattermost-server/model/websocket_client.go:99 +0x3f
```

**To Reproduce**
1. Add custom CA certificate to /usr/local/share/ca-certificates
2. Run docker container with `/bin/sh -c "apk add --update --no-cache ca-certificates && update-ca-certificates --fresh && /go/bin/botkube" `

**Additional context**
Run with defaults:
```
/bin/sh -c "/go/bin/botkube"
```
```
INFO[2019-04-16T06:57:01Z] Allowed Events - map[{Resource:clusterrole Namespace:all}:true {Resource:configmap Namespace:all}:true {Resource:daemonset Namespace:all}:true {Resource:deployment Namespace:all}:true {Resource:ingress Namespace:all}:true {Resource:job Namespace:all}:true {Resource:namespace Namespace:all}:true {Resource:node Namespace:all}:true {Resource:persistentvolume Namespace:all}:true {Resource:persistentvolumeclaim Namespace:all}:true {Resource:pod Namespace:all}:true {Resource:role Namespace:all}:true {Resource:rolebinding Namespace:all}:true {Resource:secret Namespace:all}:true {Resource:service Namespace:all}:true]
DEBU[2019-04-16T06:57:01Z] Registering the filter&{}
DEBU[2019-04-16T06:57:01Z] Registering the filter&{}
DEBU[2019-04-16T06:57:01Z] Registering the filter&{}
INFO[2019-04-16T06:57:01Z] Starting controller
INFO[2019-04-16T06:57:01Z] Starting mattermost bot
ERRO[2019-04-16T06:57:06Z] There was a problem pinging the Mattermost server URL: https://private.server
Error: https://private.server/api/v4/config/client?format=old: model.client.connecting.app_error, Get https://private.server/api/v4/config/client?format=old: x509: certificate signed by unknown authority
ERRO[2019-04-16T06:57:06Z] Error in connecting to Mattermost team TEAM
Error: https://private.server/api/v4/teams/name/TEAM: model.client.connecting.app_error, Get https://private.server/api/v4/teams/name/TEAM: x509: certificate signed by unknown authority

```

### PR DESCRIPTION
**Pull request by arush-sal**
Tested for self signed certificates in Mattermost

Closes #81

### CODE REVIEW

**Comment by PrasadG193** [comment]
@mugdha-adhav Do we need to add CA certificate while deploying BotKube backend?

File: README.md [hunk 1]
```diff
@@ -78,6 +78,8 @@ $ kubectl create ns botkube && kubectl create -f deploy-all-in-one.yaml -n botku
 
 - Check pod status in botkube namespace. Once running, send **@BotKube ping** in the Slack channel to confirm if BotKube is responding correctly.
 
+- To deploy with TLS, download and use deploy-all-in-one-tls.yaml. Replace **ENCODED_CERTIFICATE** with your base64 encoded certificate value in the secret.
+
```
Comment (PrasadG193) [inline README.md]: I think we should provide a command for encoding cert, something like -
`cat cert.pem | base64`

**Comment by PrasadG193** [comment]
Fixes: https://github.com/infracloudio/botkube/issues/81

File: README.md (hunk 1 above)
Comment (mugdha-adhav) [reply README.md]: Yes will update

**Comment by PrasadG193** [comment]
@mugdha-adhav Could you also update related docs and raise PR against botkube-docs develop branch?

**Comment by PrasadG193** [comment]
@mugdha-adhav can you please rebase with develop?

**Comment by arush-sal** [comment]
Close #81

**Pull request merged**